pinner/pinning_op/pinning_op.go

```diff
@@ -0,0 +1,77 @@
+package pinning_op
```

Comment:
also wondering why this has to be its own package, seems more convenient to have it be part of the pinner package